Modules/xASL_module_Structural.m

```diff
@@ -185,14 +185,30 @@
             xASL_io_SaveNifti(x.P.Path_T1, x.P.Path_T1, IM(:,:,:,1,1,1,1), [], false);
         end
     end
-    if xASL_exist(x.P.Path_FLAIR, 'file') % Fix multiple T1w iterations
+    if xASL_exist(x.P.Path_FLAIR, 'file') % Fix multiple FLAIR iterations
         IM = xASL_io_Nifti2Im(x.P.Path_FLAIR);
         if size(IM,4)>1 || size(IM,5)>1 || size(IM,6)>1 || size(IM,7)>1
-            warning(['Too many dims, using first: ' x.P.Path_T1]);
+            warning(['Too many dims, using first: ' x.P.Path_FLAIR]);
             xASL_Move(x.P.Path_FLAIR, x.P.Path_FLAIR_ORI, true);
             xASL_io_SaveNifti(x.P.Path_FLAIR, x.P.Path_FLAIR, IM(:,:,:,1,1,1,1), [], false);
         end
-    end
+	end
```

Comment:
Here we iterate the same code multiple times, better to iterate over {x.P.Path_FLAIR x.P.Path_T1 x.P.Path_T2 ...}

Same below; to avoid the code growing out of hand

Reply:
Good point. Fixed